helix-core/src/main/java/org/apache/helix/manager/zk/CallbackHandler.java

```diff
@@ -200,6 +206,45 @@ protected void handleEvent(NotificationContext event) {
         logger.warn("Exception in callback processing thread. Skipping callback", e);
       }
     }
+
+    public boolean queueIsEmpty() {
+      return _eventQueue.isEmpty();
+    }
+  }
+
+  /**
+   * This class represents each periodic refresh task
+   * If the deadline passed, enqueue an event to do a refresh
+   * If the deadline has not passed (some events happened in between so don't need to do a refresh),
+   * wait until reaching the interval and check again
+   */
+  class TriggerTask implements Runnable {
+    @Override
+    public void run() {
+      try {
+        while (true) {
+          long currentTime = System.currentTimeMillis();
+          long remainingTime = _lastInvokeTime + _periodicTriggerInterval - currentTime;
```

Comment:
To reduce the usage of _periodicTriggerInterval in the whole callback class, could you please pass the interval as a constructor parameter? That would be a cleaner design.

Reply:
Updated.